core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/AttributeComponent.scala

```diff
@@ -320,20 +320,27 @@ trait AttributeComponent {
       }
     }
 
-    def findUniqueStringsByNameQuery(attrName: AttributeName, queryString: Option[String]) = {
+    def findUniqueStringsByNameQuery(attrName: AttributeName, queryString: Option[String], limit: Option[Int] = None) = {
 
       val basicFilter = filter(rec =>
         rec.namespace === attrName.namespace &&
           rec.name === attrName.name &&
           rec.valueString.isDefined)
 
-      val res = (queryString match {
+      var res = (queryString match {
```

Comment:
Instead of making this a `var` (mutable values are generally frowned upon in Scala), I would recommend building the query like:

```
      val baseQuery = (queryString match {
        case Some(query) => basicFilter.filter(_.valueString.like(s"%$query%"))
        case None => basicFilter
      }).groupBy(_.valueString).map(queryThing =>
        (queryThing._1, queryThing._2.length)
      ).sortBy(r => (r._2.desc, r._1))

      val res = limit match {
        case Some(limit) => baseQuery.take(limit)
        case None => baseQuery
      }

      res.map(x => (x._1.get, x._2))
```

Reply:
Done in [e5c8eac](https://github.com/broadinstitute/rawls/pull/1663/commits/e5c8eac13721100800c30e99fd8964fd7c6eca6b)